### PR TITLE
Migrate realms if configured to use RH-SSO themes

### LINKED ISSUE #17484: Migrate realms if configured to use RH-SSO themes
### Description

In Keycloak 22 there will no longer be RH-SSO themes. This may result in an issue when customers are upgrading to Keycloak 22 from RH-SSO if they have explicitly used the RH-SSO themes for realms or clients.

We should add a migration to replace RH-SSO themes with Keycloak themes.

Done for the new admin console already:
https://github.com/keycloak/keycloak/blob/main/model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo21_0_0.java

We'll need to do the same for login and new account console.

For realms we can do it as above, but for clients that may be more complicated as we can't really iterate over all the clients in all realms.

This is only relevant for the old store, so we can consider doing an custom SQL migration in Liquibase that will be more efficient.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by rmartinc**
Closes https://github.com/keycloak/keycloak/issues/17484

Removing login and account themes if they are `rh-sso` to use the default one. I understand that admin and email are not needed. Let me know if they should be modified too.

A custom liquibase task for version 22 is used to perform the changes. Liquibase was needed for clients, so I decided to use the same class for realms too, without adding an extra migration task.

The migration tests are functional now (most part of this PR). Tested with the five profiles (mariadb, mysql, postgres, mssql and oracle). As commented in the meeting I needed to start from 19.0.3 because of differences in the boot options (for example `--http-relative-path` is missing in previous versions). The README has been modified with the new details.

The test imports and starts the old version (initializing db to 19.0.3), then start the new version (upgrading to last version) and perform the checks. The startup of the previous version is not really necessary (the import already does the db initialization before importing the realms) although it tests that previous version boots OK. If you think this modification is a good idea, I would create a new issue for it.

### CODE REVIEW

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    protected void generateStatementsImpl() throws CustomChangeException {
+        // remove login theme for realms
+        statements.add(new UpdateStatement(null, null, database.correctObjectName("REALM", Table.class))
+                .addNewColumnValue("LOGIN_THEME", null)
+                .setWhereClause("LOGIN_THEME=?")
+                .addWhereParameter("rh-sso"));
+        // remove account theme for realms
+        statements.add(new UpdateStatement(null, null, database.correctObjectName("REALM", Table.class))
```
Comment (mposolda) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java:40]: Just looked at the RH-SSO 7.6.3 and seeing that "email" theme has also "rh-sso" as available value. So looks that one should be also modified the same way? It was not mentioned in the description of the GH issue, but now figured that we have it according to this screenshot.

AFAIK admin theme is not needed as it was already done in `MigrateTo21_0_0`

**Review by mposolda — COMMENTED** [review]
LGTM, but I guess we need also email theme? I've quickly checked in RH-SSO 7.6.3 and it has "rh-sso" as email theme possibility.

@hmlnarik Could you please review as well or delegate to someone from "store" team?

@miquelsi Can you please review the changes related to MigrationTest? It seems that DB migrationTest is broken in current `main` and this PR fixes it at least for migration from Keycloak 19.0.3 (the 1st version using quarkus distribution). Does the changes look ok to you? It seems that for testing real migration from RH-SSO, we may need also support for migration from Keycloak 18.x (that corresponds to RH-SSO), but adding that is out of scope of this PR though. But hopefully at least changes in this PR can help with that and the Keycloak 18.X can be done as a follow-up. WDYT?

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java (hunk 1 above)
Comment (rmartinc) [reply model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java:40]: Yes, email and admin also have rh-sso version. The admin one was already changed by a previous upgrade but email is not managed. Reading the description I thought it was not needed. Okis, I'll add another update sentence then for email.

**Review by miquelsi — DISMISSED** [review]
Thanks for fixing this!

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java (hunk 1 above)
Comment (rmartinc) [reply model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java:45]: Okis! Done!

Comment (mposolda) [reply model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate22_0_0_RemoveRhssoThemes.java:45]: Thanks!

**Review by mposolda — APPROVED** [review]
@miquelsi Thanks for the review

**Review by vramik — APPROVED** [review]
Thanks @rmartinc for the PR. LGTM

**Pull request merged**